src/menu/menu.c

```diff
@@ -253,46 +182,34 @@ item_create(struct menu *menu, const char *text, bool show_arrow)
 	/* Font nodes */
 	menuitem->normal.buffer = scaled_font_buffer_create(menuitem->normal.tree);
 	menuitem->selected.buffer = scaled_font_buffer_create(menuitem->selected.tree);
-	if (!menuitem->normal.buffer || !menuitem->selected.buffer) {
-		wlr_log(WLR_ERROR, "Failed to create menu item '%s'", text);
-		/*
-		 * Destroying the root node will destroy everything,
-		 * including the node descriptor and scaled_font_buffers.
-		 */
-		wlr_scene_node_destroy(&menuitem->tree->node);
-		free(menuitem);
-		return NULL;
-	}
+	assert(menuitem->normal.buffer && menuitem->selected.buffer);
```

Comment:
From memory this has been added to catch using a font size of `0` but I didn't verify.
Also, some recent refactoring seems to have broken `lab_data_buffer` with either height or width of `0` as it now asserts in `data_buffer_begin_data_ptr_access()` whereas before it would return `NULL` and thus trigger this check.

Reply:
I see. I thought it was just for OOM and didn't think about that. But when would anyone want to set the font sizes to zero? Can't we just force them to be positive?

Reply:
I think a font size of 0 was used at some to hide the title (or even the whole titlebar, can't remember right now). Personally I don't feel strongly about supporting a font size of a `0` vs enforcing a font size `> 0`. If we go the 2nd route we should still add an assert when creating a lab_data_buffer though.